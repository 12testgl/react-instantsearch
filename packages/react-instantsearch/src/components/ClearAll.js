import PropTypes from 'prop-types';
import React, { Component } from 'react';

import translatable from '../core/translatable';

class ClearAll extends Component {
  static propTypes = {
    cx: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    refine: PropTypes.func.isRequired,
  };

  render() {
    const { cx, translate, items, refine } = this.props;
    const isDisabled = items.length === 0;

    return (
      <button
        className={cx('button', isDisabled && 'button--disabled')}
        onClick={() => refine(items)}
        disabled={isDisabled}
      >
        {translate('reset')}
      </button>
    );
  }
}

export default translatable({
  reset: 'Clear all filters',
})(ClearAll);
