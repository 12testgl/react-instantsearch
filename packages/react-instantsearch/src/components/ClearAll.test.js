/* eslint-env jest, jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import ClearAll from './ClearAll';

describe('ClearAll', () => {
  it('renders a clickable button', () =>
    expect(
      renderer
        .create(
          <ClearAll
            refine={() => null}
            items={[{ filter: 1 }]}
            cx={(...x) => x.join(' ')}
          />
        )
        .toJSON()
    ).toMatchSnapshot());

  it('has a disabled mode', () =>
    expect(
      renderer
        .create(
          <ClearAll refine={() => null} items={[]} cx={(...x) => x.join(' ')} />
        )
        .toJSON()
    ).toMatchSnapshot());

  it('is disabled when there is no filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearAll refine={refine} items={[]} cx={(...x) => x.join(' ')} />
    );

    const btn = wrapper.find('button');
    expect(refine.mock.calls).toHaveLength(0);
    btn.simulate('click');
    expect(refine.mock.calls).toHaveLength(0);
  });

  it('is not disabled when there are filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearAll
        refine={refine}
        items={[{ value: 'test', label: 'test: test' }]}
        cx={(...x) => x.join(' ')}
      />
    );

    const btn = wrapper.find('button');
    expect(refine.mock.calls).toHaveLength(0);
    btn.simulate('click');
    expect(refine.mock.calls).toHaveLength(1);
  });
});
