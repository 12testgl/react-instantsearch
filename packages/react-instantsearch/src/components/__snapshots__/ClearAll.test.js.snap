// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearAll has a disabled mode 1`] = `
<button
  className="button button--disabled"
  disabled={true}
  onClick={[Function]}
>
  Clear all filters
</button>
`;

exports[`ClearAll renders a clickable button 1`] = `
<button
  className="button false"
  disabled={false}
  onClick={[Function]}
>
  Clear all filters
</button>
`;
